operator/pkg/charts/reconcile.go

```diff
@@ -0,0 +1,134 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1beta3 "github.com/k0sproject/k0s/pkg/apis/helm/v1beta1"
+	v1beta2 "github.com/k0sproject/k0s/pkg/apis/k0s/v1beta1"
+	"github.com/replicatedhq/embedded-cluster/kinds/apis/v1beta1"
+	"github.com/replicatedhq/embedded-cluster/operator/pkg/release"
+	"sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ReconcileHelmCharts reconciles the helm charts from the Installation metadata with the clusterconfig object.
+func ReconcileHelmCharts(ctx context.Context, cli client.Client, in *v1beta1.Installation) error {
```

Comment:
this is unmodified in terms of logic, just moved and changed to take a client as an argument